consensus/helper/helper.go

```diff
@@ -313,9 +313,32 @@ func (h *Helper) SkipTo(tag uint64, id []byte, peers []*pb.PeerID) {
 	if h.valid {
 		logger.Warning("State transfer is being called for, but the state has not been invalidated")
 	}
-	info := &pb.BlockchainInfo{}
-	proto.Unmarshal(id, info)
-	h.sts.AddTarget(info.Height-1, info.CurrentBlockHash, peers, tag)
+
+	// This looks racey, but this should always be called in a serial fashion so it should not be, also this will be removed when the executor is introduced
+	// This is just a temporary hack to enable legacy-like behavior until the executor is finished
+	if !h.stateTransfering {
+		h.stateTransfering = true
+		info := &pb.BlockchainInfo{}
+		proto.Unmarshal(id, info)
+		go func() {
+			h.Initiated(info.Height-1, info.CurrentBlockHash, peers, tag)
+
+			for {
```

Comment:
Ah! Good catch. State transfer will benignly ignore transfer to targets it already has, but this will leak a go routine and needlessly waste cycles, I'll pick a fix shortly.  This entire codepath should be going away shortly though, which is why there is nothing like an accompanying test case to catch this.